agent/src/main/scala/za/co/absa/atum/agent/AtumAgent.scala

```diff
@@ -18,13 +18,15 @@ package za.co.absa.atum.agent
 import com.typesafe.config.{Config, ConfigFactory}
 import za.co.absa.atum.agent.AtumContext.AtumPartitions
 import za.co.absa.atum.agent.dispatcher.{ConsoleDispatcher, HttpDispatcher}
-import za.co.absa.atum.model.dto.{CheckpointDTO, PartitioningDTO}
+import za.co.absa.atum.model.dto.{CheckpointDTO, PartitioningSubmitDTO}
 
 /**
  * Entity that communicate with the API, primarily focused on spawning Atum Context(s).
  */
 class AtumAgent private[agent] () {
 
+  private[this] var contexts: Map[AtumPartitions, AtumContext] = Map.empty
```

Comment:
Because having such an important attribute in the bottom of this class, where it's almost invisible, is really weird. 

Having vals/vars on the bottom, and methods first, is odd habit that leads to harder to read code in my personal opinion. I've never saw this in the past in any programming language. Even Java has some recommendations and it's not like this, on SO it's being mentioned: https://softwareengineering.stackexchange.com/questions/380771/in-which-order-functions-should-be-defined-scala-methods. This is usually never enforced and not a 'best practice', but methods are usually the last ones in a class, class variables first

Reply:
Interesting I was thought and number of my mentors was of the same opinion that in front are the public member of the class, then protected and private ones at the end.
Anyway, not worth further discussion, definitely not now 😉